testsuite/shared/src/main/java/org/jboss/as/test/shared/util/AssumeTestGroupUtil.java

```diff
@@ -241,7 +244,15 @@ public static boolean isFullDistribution() {
      * @throws AssumptionViolatedException if a docker client cannot be initialized
      */
     public static void assumeDockerAvailable() {
-        assumeCondition("Docker is not available.", AssumeTestGroupUtil::isDockerAvailable);
+        try {
+            assumeCondition("Docker is not available.", AssumeTestGroupUtil::isDockerAvailable);
+        } catch (AssumptionViolatedException ex) {
+            if (System.getProperty("org.wildfly.test.require.docker").equals("true")){
```

Comment:
I kicked off a discussion at https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/WFLY-18258.20--.20handling.20docker.20being.20unavailable.20in.20CI re whether we should do it like this or invert it. Absent any consensus to change this, this is fine.